packages/core/src/tracer.ts

```diff
@@ -0,0 +1,60 @@
+import type { Plugin, PromiseOrValue, Tracer } from '@envelop/types';
+import { isPromise, mapMaybePromise } from './utils';
+
+export function getTracer<PluginsContext extends Record<string, any>>(
+  plugins: Plugin[],
+): Tracer<PluginsContext> | undefined {
+  let tracer: Tracer<PluginsContext> | undefined;
+  for (const plugin of plugins) {
+    if (plugin.tracer) {
+      if (tracer) {
+        throw new Error('A plugin has already declared a tracer. Only one tracer is allowed');
+      }
+      tracer = plugin.tracer;
+    }
+  }
+  return tracer;
+}
+
+export const getTraced = <TResult, TArgs extends any[]>(context: any) => ({
+  sync(
```

Comment:
nit: `sync` feels like `synchronize` when it's being called like that. 
maybe `traceSyncFn` or something?

Reply:
what about `fn` and `asyncFn` ? Because you already have the `traced` part in the variable name at usage:
```ts
cons traced = getTraced(context)
const tracedFn = traced.fn(fn)
const asyncTracedFn = traced.asyncFn(asyncFn)
```